cmd/search_log.go

```diff
@@ -137,3 +179,129 @@ func clicksFn(cmd *cobra.Command, args []string) {
 	}
 	printCsv(records)
 }
+
+func latencyFn(cmd *cobra.Command, args []string) {
+	logger := initLogger()
+	headers := []string{
+		"#", "SearchId", "Term",
+	}
+	headers = append(headers, consts.LATENCY_LOG_OPERATIONS_FOR_SEARCH...)
+	if csvFile != "" {
+		appendCsvToFile(csvFile, [][]string{headers})
+	} else {
+		printCsv([][]string{headers})
+	}
+	totalQueries := 0
+	SLICES := 100
+	for i := 0; i < SLICES; i++ {
+		s := elastic.NewSliceQuery().Id(i).Max(SLICES)
+		queries, err := logger.GetLattestQueries(s, null.StringFrom("now-7d/d"), null.BoolFrom(false))
+		utils.Must(err)
+		totalQueries += len(queries)
+		sortedQueries := make(search.CreatedSearchLogs, 0, len(queries))
+		for _, q := range queries {
+			sortedQueries = append(sortedQueries, q)
+		}
+		sort.Sort(sortedQueries)
+		records := [][]string{}
+		for i, sl := range sortedQueries {
+			utils.Must(err)
+			var latencies []string
+			for _, op := range consts.LATENCY_LOG_OPERATIONS_FOR_SEARCH {
+				hasOp := false
+				for _, tl := range sl.ExecutionTimeLog {
+					if tl.Operation == op {
+						latancy := strconv.FormatInt(tl.Time, 10)
+						latencies = append(latencies, latancy)
+						hasOp = true
+						break
+					}
+				}
+				if !hasOp {
+					latencies = append(latencies, "0")
+				}
+			}
+			record := []string{
+				fmt.Sprintf("%d", i+1),
+				sl.SearchId,
+				sl.Query.Term,
+			}
+			record = append(record, latencies...)
+			records = append(records, record)
+		}
+		if csvFile != "" {
+			appendCsvToFile(csvFile, records)
+		} else {
+			printCsv(records)
+		}
+	}
+	log.Infof("Found %d queries.", totalQueries)
+}
+
+func latencyAggregateFn(cmd *cobra.Command, args []string) {
+
+	const metaColumnsCnt = 3                  //  "#","SearchId","Term"
```

Comment:
Len of constant slice (keep the constant in this file).

Reply:
constant slice is not supported in go

Reply:
changed to len(latencyMetaHeaders)